libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -626,17 +626,21 @@ void AP_GPS::update_instance(uint8_t instance)
     // detection to run again
     if (!result) {
         if (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS) {
-            // free the driver before we run the next detection, so we
-            // don't end up with two allocated at any time
-            delete drivers[instance];
-            drivers[instance] = nullptr;
             memset(&state[instance], 0, sizeof(state[instance]));
             state[instance].instance = instance;
-            state[instance].status = NO_GPS;
             state[instance].hdop = GPS_UNKNOWN_DOP;
             state[instance].vdop = GPS_UNKNOWN_DOP;
-            timing[instance].last_message_time_ms = tnow;
```

Comment:
I apologize dreadfully, but this line does need to be kept. By removing this you stop all the vehicles from logging that the GPS is actually no longer present.